### PR TITLE
Move VSTS extension to its own module

### PR DESCRIPTION
**Pull request by gregaubert**

### CODE REVIEW

File: server/sonar-vsts/package.json [hunk 1]
```diff
@@ -14,51 +14,48 @@
...
   "devDependencies": {
     "@types/classnames": "2.2.3",
-    "@types/enzyme": "3.1.9",
-    "@types/jest": "22.2.0",
-    "@types/lodash": "4.14.104",
-    "@types/react": "16.0.40",
-    "@types/react-dom": "16.0.4",
-    "autoprefixer": "8.1.0",
-    "awesome-typescript-loader": "4.0.1",
+    "@types/enzyme": "3.1.6",
```
Comment (stas-vilchik) [inline server/sonar-vsts/package.json]: what's the reason of this downgrade?

File: server/sonar-vsts/package.json [hunk 2]
```diff
@@ -14,51 +14,48 @@
...
+    "babel-loader": "7.1.2",
+    "babel-plugin-dynamic-import-node": "1.1.0",
     "babel-plugin-syntax-dynamic-import": "6.18.0",
-    "babel-plugin-transform-class-properties": "6.24.1",
+    "babel-plugin-transform-class-properties": "^6.22.0",
     "babel-plugin-transform-object-rest-spread": "6.26.0",
-    "babel-plugin-transform-react-constant-elements": "6.23.0",
-    "babel-plugin-transform-react-jsx-self": "6.22.0",
-    "babel-plugin-transform-react-jsx-source": "6.22.0",
+    "babel-plugin-transform-react-constant-elements": "^6.22.0",
```
Comment (stas-vilchik) [inline server/sonar-vsts/package.json]: use strict versions

File: server/sonar-vsts/package.json [hunk 3]
```diff
@@ -118,17 +114,5 @@
     "jsxBracketSameLine": true,
     "printWidth": 100,
     "singleQuote": true
-  },
-  "lint-staged": {
```
Comment (stas-vilchik) [inline server/sonar-vsts/package.json]: don't you need this?

File: server/sonar-vsts/scripts/utils/getMessages.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+const fs = require('fs');
```
Comment (stas-vilchik) [inline server/sonar-vsts/scripts/utils/getMessages.js]: I don't think you need this file/functionality

File: server/sonar-vsts/scripts/utils/formatSize.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+module.exports = function(bytes) {
```
Comment (stas-vilchik) [inline server/sonar-vsts/scripts/utils/formatSize.js]: I suggest to drop this file/functionality to reduce the amount of configuration/scripts

File: server/sonar-vsts/scripts/analyze.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+/* eslint-disable no-console */
+process.env.NODE_ENV = 'production';
```
Comment (stas-vilchik) [inline server/sonar-vsts/scripts/analyze.js]: let's drop this analyze script

File: server/sonar-vsts/.eslintignore [hunk 7]
```diff
@@ -0,0 +1 @@
+src/main/js/libs
```
Comment (stas-vilchik) [inline server/sonar-vsts/.eslintignore]: you don't need this file

File: server/sonar-vsts/.eslintrc [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+  "parserOptions": {
+    "ecmaVersion": 6,
+    "sourceType": "module",
+    "ecmaFeatures": {
+      "jsx": true,
+      "modules": true
+    }
+  },
+
+  "globals": {
```
Comment (stas-vilchik) [inline server/sonar-vsts/.eslintrc]: remove this section

File: server/sonar-vsts/tsconfig.json [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    "noUnusedParameters": true,
+    "noImplicitAny": true,
+    "strict": true,
+    "strictFunctionTypes": false,
+    "target": "es5",
+    "jsx": "react",
+    "lib": ["es2017", "dom"],
+    "module": "esnext",
+    "moduleResolution": "node",
+    "typeRoots": ["./node_modules/@types"]
```
Comment (stas-vilchik) [inline server/sonar-vsts/tsconfig.json]: you can safely remove this line

File: server/sonar-vsts/tsconfig.json [hunk 10]
```diff
@@ -0,0 +1,18 @@
+{
+  "compilerOptions": {
+    "allowJs": true,
+    "checkJs": false,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "noImplicitAny": true,
+    "strict": true,
+    "strictFunctionTypes": false,
```
Comment (stas-vilchik) [inline server/sonar-vsts/tsconfig.json]: I think you can drop this line

File: server/sonar-vsts/package.json [hunk 11]
```diff
@@ -0,0 +1,118 @@
+{
+  "name": "SonarQube-VSTS",
```
Comment (stas-vilchik) [inline server/sonar-vsts/package.json]: - lowercase the name
- add `"private": true`

**Comment by stas-vilchik** [comment]
So have you tried to `yarn add` the `VSS.SDS`? Did it work?

File: server/sonar-vsts/package.json (hunk 1 above)
Comment (gregaubert) [reply server/sonar-vsts/package.json]: Having same versions as in sonar-web

File: server/sonar-vsts/package.json (hunk 2 above)
Comment (gregaubert) [reply server/sonar-vsts/package.json]: I'll update sonar-web too then

**Pull request merged**